narwhals/_pandas_like/namespace.py

```diff
@@ -250,3 +251,118 @@ def concat(
                 backend_version=self._backend_version,
             )
         raise NotImplementedError
+
+    def when(
+        self,
+        *predicates: IntoPandasLikeExpr | Iterable[IntoPandasLikeExpr],
+    ) -> PandasWhen:
+        plx = self.__class__(self._implementation, self._backend_version)
+        if predicates:
+            condition = plx.all_horizontal(*flatten(predicates))
+        else:
+            msg = "at least one predicate needs to be provided"
+            raise TypeError(msg)
+
+        return PandasWhen(condition, self._implementation, self._backend_version)
+
+
+def _when_then_value_arg_process(
+    plx: PandasLikeNamespace,
+    value: PandasLikeExpr | PandasLikeSeries | Any,
+    *,
+    shape: tuple[int],
+) -> PandasLikeExpr:
+    from narwhals.dependencies import get_numpy
+
+    np = get_numpy()
+
+    if not np:
+        msg = "numpy is required for this function"
+        raise ImportError(msg)
+    if isinstance(value, PandasLikeExpr):
+        return value
+    elif isinstance(value, PandasLikeSeries):
+        return plx._create_expr_from_series(value)
+    elif isinstance(value, np.ndarray):
```

Comment:
indeed, if `value` is a numpy array, it implies that numpy was already imported

Reply:
good point, didn't look into cudf much actually, will have to check it out in more detail

Reply:
I put the error throw path just in case that someone or something somehow messed with the python installation path and `numpy` was unavailable at runtime. This is quite an esoteric scenario but still put the code there just in case. Can delete the failure
path and simplify complexity.

`numpy` is indeed a required dependency of pandas, which js why in the tests I had to mock the `get_numpy` function

If we delete the failure path we can also simplify the test module

Reply:
I actually just saw the changes @MarcoGorelli that you just did what I mentioned in the previous comment

oops on my part for not checking the last commit